bot/cogs/team.py

```diff
@@ -0,0 +1,346 @@
+from random import shuffle
+
+from discord import Member, VoiceChannel, errors
+from discord.ext import commands, tasks
+from common.discord_helper import get_member_list, string_to_role
+
+from common.bot_logger import get_logger
+from common.helper_functions import reply_and_delete
+
+logger = get_logger(__name__)
+
+
+class Team:
+    def __init__(self, name: str):
+        self.__name = name
+        self.__members = []
+        self.__size = 0
+        self.__vc = None
+
+    @property
+    def name(self):
+        return self.__name
+
+    @name.setter
+    def name(self, name: str):
+        self.__name = name
+
+    @property
+    def size(self):
+        return self.__size
+
+    @property
+    def members(self):
+        return self.__members
+
+    def add_member(self, member: Member):
+        if member not in self.__members:
+            self.__members.append(member)
+            self.__size += 1
+
+    def rem_member(self, member: Member):
+        for m in self.__members:
+            if m.id == member.id:
+                self.__members.remove(m)
+                self.__size -= 1
+
+    @property
+    def vc(self):
+        return self.__vc
+
+    @vc.setter
+    def vc(self, vc: VoiceChannel):
+        self.__vc = vc
+
+
+class TeamCog(commands.Cog, name="Team commands"):
+    def __init__(self, bot):
+        self.bot = bot
+        self.teams = []
+        logger.info("Team cog initialized")
+        self.clean_up_vc.start()
+
+    def create_team(self, team, name):
+        i = 2
+        new_name = name
+        while new_name in [t.name for t in self.teams]:
+            new_name = name + "." + str(i)
+            i += 1
+        teamObj = Team(new_name)
+        self.teams.append(teamObj)
+
+        for member in team:
+            for t in self.teams:
+                for m in t.members:
+                    if member.id == m.id:
+                        t.rem_member(member)
+            teamObj.add_member(member)
+
+    @commands.group(
+        name="team", invoke_without_command=True, help="Get help for team commands"
+    )
+    async def team(self, ctx):
+        await reply_and_delete(ctx, "'team' needs a sub-command!")
+        await ctx.send_help(self.team)
+
+    @team.command(help="Create a new team using: !team create [name(s)]")
+    @commands.has_permissions(administrator=True)
```

Comment:
I would use `manage_channels` as the permission required to do this. I believe that should mean that as long as you have permissions on the text channel it will execute for the voice channel you are in

So there should probably also be a check if you have manage_channels permission on the vc.